shared/src/main/scala/amf/core/remote/Context.scala

```diff
@@ -1,93 +1,102 @@
-      case RelativeToIncludedFile(_) => Context.stripFile(current, platform.operativeSystem())
+    val base = url match {
+      case Absolute(_)               => None
+      case RelativeToRoot(_)         => Some(root)
+      case RelativeToIncludedFile(_) => Some(current)
     }
-    concatWithoutDuplicateSlash(prefix, url)
-  }
-
-  private def concatWithoutDuplicateSlash(prefix: String, url: String) = {
-    if (prefix.nonEmpty && prefix.last == '/' && url.nonEmpty && url.head == '/') prefix + url.drop(1)
-    else prefix + url
+    resolveRelativeTo(base, url)
   }
 
-  private def applyMapping(path: String): String =
-    mappings.find(m => path.startsWith(m._1)).fold(path)(m => path.replace(m._1, m._2))
 }
 
 object Context {
   private def apply(platform: Platform,
                     history: List[String],
-                    currentResolved: String,
-                    mapping: Map[String, String]): Context =
-    new Context(platform, history :+ currentResolved, mapping)
-
-  def apply(platform: Platform, root: String): Context = Context(platform, root, Map.empty)
+                    currentResolved: String): Context =
+    new Context(platform, history :+ currentResolved)
 
-  def apply(platform: Platform): Context = Context(platform, "", Map.empty)
+  def apply(platform: Platform): Context = new Context(platform, Nil)
 
-  def apply(platform: Platform, root: String, mapping: Map[String, String]): Context =
-    new Context(platform, Option(root).filter(_.nonEmpty).map(List(_)).getOrElse(Nil), mapping)
+  def apply(platform: Platform, root: String): Context =
+    new Context(platform, List(root))
 
-  def apply(platform: Platform, mapping: Map[String, String]): Context =
-    new Context(platform, Nil, mapping)
+  def stripFileName(url: String): String = {
+    val withoutFrag = if (url.contains("#")) url.split("#").head else url
 
-  private def stripFile(url: String, so: String): String = {
-    val containsBackSlash = url.contains('\\') && so == "win"
-    val containsForwardSlash = url.contains('/')
+    val so = platform.operativeSystem()
+    val containsBackSlash = withoutFrag.contains('\\') && so == "win"
+    val containsForwardSlash = withoutFrag.contains('/')
     if (!containsBackSlash && !containsForwardSlash) {
       return ""
     }
     val sep = if (containsBackSlash) '\\' else '/'
-    val lastPieceHasExtension = url.split(sep).last.contains('.')
+    val lastPieceHasExtension = withoutFrag.split(sep).last.contains('.')
     if (lastPieceHasExtension) {
-      url.substring(0, url.lastIndexOf(sep) + 1)
-    } else if (!url.endsWith(sep.toString)) {
-      url + sep
+      withoutFrag.substring(0, withoutFrag.lastIndexOf(sep) + 1)
+    } else if (!withoutFrag.endsWith(sep.toString)) {
+      withoutFrag + sep
     } else {
-      url
+      withoutFrag
     }
   }
+
+  def resolveRelativeTo(base: Option[String], url: String): String = {
```

Comment:
URL o URI.resolve can do this I think

Reply:
URI.resolve would be ideal, however this would require some preprocessing for the base and url. Many cases came up when giving this a try.